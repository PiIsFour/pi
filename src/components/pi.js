import React from 'react'
import Store from './store'
import reducerFactory from './pi.reducer'
import {create} from './pi.actions'
import dist from '../model/dist'

function Card ({children, width = 1, height = 1, style = {}}) {
	return <div style={{
		...style,
		gridColumn: 'span ' + width,
		gridRow: 'span ' + height,
		backgroundColor: '#424242',
		boxShadow: '2px 2px 5px 0px #000000'
	}}>
		{children}
	</div>
}

export function Pi ({state, dispatch}) {
	return <div id="pi" style={{
		display: 'grid',
		gridTemplateColumns: 'repeat(auto-fit, minmax(200px, 1fr))',
		gridGap: '10px'
	}}>
		{!state && <Card style={{textAlign: 'center'}}>
			<a className="button" onClick={() => dispatch(create())}>Start</a>
		</Card>}
	</div>
}

export default function () {
	return <Store reducer={reducerFactory(dist)}>
		<Pi />
	</Store>
}
