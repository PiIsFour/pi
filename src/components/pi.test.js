import React from 'react'
import { shallow } from 'enzyme'
import enzymeToJson from 'enzyme-to-json'

import {Pi} from './pi'

describe('pi', () => {
	it('snapshot matches', () => {
		const wrapper = shallow(<Pi />)
		expect(enzymeToJson(wrapper)).toMatchSnapshot()
	})

	it('shows start button if state is undefined', () => {
		const wrapper = shallow(<Pi />)
		expect(wrapper.find('.button')).toExist()
		expect(wrapper.find('.button')).toHaveText('Start')
	})

	it('dispaches the create action when start button is clicked', () => {
		const mockDispatch = jest.fn()
		const wrapper = shallow(<Pi dispatch={mockDispatch}/>)
		wrapper.find('.button').simulate('click')
		expect(mockDispatch).toBeCalledWith({type: 'CREATE'})
	})
})
