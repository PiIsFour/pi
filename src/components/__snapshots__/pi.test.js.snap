// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pi snapshot matches 1`] = `
<div
  id="pi"
  style={
    Object {
      "display": "grid",
      "gridGap": "10px",
      "gridTemplateColumns": "repeat(auto-fit, minmax(200px, 1fr))",
    }
  }
>
  <Card
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <a
      className="button"
      onClick={[Function]}
    >
      Start
    </a>
  </Card>
</div>
`;
